### DISCUSSION TITLE
Does autoLLM support HuggingFace LLM's

### DISCUSSION THREAD

**Discussion by 0xSaurabhx** [safevideo/autollm#96] [Q&A] [answered]
need to integrating huggingface models also other than just proprietary LLM's.

**Comment by talhasarit** [comment]
Thank you for highlighting the need for HuggingFace model integration. We value your feedback! In response to your suggestion, we've updated our README with examples demonstrating the integration of HuggingFace models. You can find the details in the [🌟 features](https://github.com/safevideo/autollm/tree/main#-features) section. Your input helps make our project better!

Best,
Talha

**Comment by ZQ-Dev8** [comment]
@SeeknnDestroy Talha, appreciate the quick update regarding HF models. Is HuggingFace's TGI supported as a backend? Specifically, if I have a HF model hosted a local TGI server, can I interact with it via AutoLLM? E.g.

```
from autollm import AutoQueryEngine

model = "meta-llama/Llama-2-7b-chat-hf"
api_base = "http://localhost:1234" #IP to TGI server

llm_params = {"model": model, "api_base": api_base, ...}
etc
```

**Comment by fcakyon** [comment]
@dcruiz01 we are working on new tutorials to make things clear to the users. We are grateful for your patience till they are published 🙏

**Comment by fcakyon** [comment, marked as answer]
> @SeeknnDestroy Talha, appreciate the quick update regarding HF models. Is HuggingFace's TGI supported as a backend? Specifically, if I have a HF model hosted a local TGI server, can I interact with it via AutoLLM? E.g.
> 
> ```
> from autollm import AutoQueryEngine
> 
> model = "meta-llama/Llama-2-7b-chat-hf"
> api_base = "http://localhost:1234" #IP to TGI server
> 
> llm_params = {"model": model, "api_base": api_base, ...}
> etc
> ```

Hello @dcruiz01, current HuggignFace examples support local and cloud TGI's. Your given code snippet should work :)
